test/e2e/flask/multichain-api/create-session.spec.ts

```diff
@@ -46,4 +56,136 @@ describe('Multichain API', function () {
       );
     });
   });
+
+  describe('Connect wallet to the multichain dapp via `externally_connectable`, call `wallet_createSession` with requested EVM scope that match the user’s enabled networks', function () {
+    it('should only select the specified EVM scopes requested by the user, and session scope contain valid accounts only', async function () {
+      await withFixtures(
+        {
+          title: this.test?.fullTitle(),
+          fixtures: new FixtureBuilder().withPopularNetworks().build(),
+          ...DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS,
+        },
+        async ({
+          driver,
+          extensionId,
+        }: {
+          driver: Driver;
+          extensionId: string;
+        }) => {
+          const INVALID_ACCOUNT = '0x9999999999999999999999999999999999999999';
+          const requestScopesToNetworkMap = {
+            'eip155:1': 'Ethereum Mainnet',
+            'eip155:42161': 'Arbitrum One',
+            'eip155:10': 'OP Mainnet',
+          };
+
+          const requestScopes = Object.keys(requestScopesToNetworkMap);
+          const networksToRequest = Object.values(requestScopesToNetworkMap);
+
+          await openMultichainDappAndConnectWalletWithExternallyConnectable(
+            driver,
+            extensionId,
+          );
+
+          await addRequestAccountsToCreateSession(driver, [
+            DEFAULT_FIXTURE_ACCOUNT,
+            INVALID_ACCOUNT,
+          ]);
+
+          await initCreateSessionScopes(driver, requestScopes);
+
+          await assertOnlyRequestedNetworksAreSelected(
+            driver,
+            networksToRequest,
+          );
+
+          await confirmAndSwitchFocusToDapp(driver);
+
+          const getSessionScopesResult = await getSessionScopes(driver);
+          for (const scope of requestScopes) {
+            /**
+             * Accounts in scope should not include invalid account {@link INVALID_ACCOUNT}, only the valid accounts.
+             */
+            const expectedSessionScope = getExpectedSessionScope(scope, [
+              DEFAULT_FIXTURE_ACCOUNT,
+            ]);
+            const result = getSessionScopesResult.sessionScopes[scope].accounts;
+
+            assert.deepEqual(
+              expectedSessionScope.accounts,
+              result,
+              `${expectedSessionScope.accounts} does not match accounts in scope ${result}`,
+            );
```

Comment:
Looking back on this, I feel like this test is already doing a lot, and we could break down this test a bit more.

Have a separate `it()` block for asserting the only the requested networks are selected in the wallet
```
          await assertOnlyRequestedNetworksAreSelected(
            driver,
            networksToRequest,
          );
```
perhaps, and the other for checking the accounts included are only the valid ones.

thoughts ?

Reply:
IMO I think we can keep these assertions in this test and just remove the assertions about networks from this test